exp/controllers/helpers.go

```diff
@@ -197,10 +199,11 @@ func AzureManagedClusterToAzureManagedControlPlaneMapper(c client.Client, log lo
 
 // MachinePoolToInfrastructureMapFunc returns a handler.ToRequestsFunc that watches for
 // MachinePool events and returns reconciliation requests for an infrastructure provider object.
-func MachinePoolToInfrastructureMapFunc(gvk schema.GroupVersionKind) handler.ToRequestsFunc {
+func MachinePoolToInfrastructureMapFunc(gvk schema.GroupVersionKind, log logr.Logger) handler.ToRequestsFunc {
```

Comment:
@devigned this one was basically the same as https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/869/files#diff-6e2cb885fbb76d5bd528cc2f3a65bcb4L312 in azuremachinepool_controller.go but without the logger so I combined them, does that make sense or was there a reason for keeping them separate?

Reply:
Makes sense. Thank you for consolidating.